src/auto/tls_interaction.rs

```diff
@@ -0,0 +1,127 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ ea993ed)
+// from gir-files (https://github.com/gtk-rs/gir-files @ 469db10)
+// DO NOT EDIT
+
+use Cancellable;
+use Error;
+#[cfg(any(feature = "v2_40", feature = "dox"))]
+use TlsCertificateRequestFlags;
+#[cfg(any(feature = "v2_40", feature = "dox"))]
+use TlsConnection;
+use TlsInteractionResult;
+use TlsPassword;
+use ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct TlsInteraction(Object<ffi::GTlsInteraction, ffi::GTlsInteractionClass>);
+
+    match fn {
+        get_type => || ffi::g_tls_interaction_get_type(),
+    }
+}
+
+pub trait TlsInteractionExt {
+    fn ask_password<'a, P: Into<Option<&'a Cancellable>>>(&self, password: &TlsPassword, cancellable: P) -> Result<TlsInteractionResult, Error>;
+
+    fn ask_password_async<'a, P: Into<Option<&'a Cancellable>>, Q: FnOnce(Result<(), Error>) + Send + 'static>(&self, password: &TlsPassword, cancellable: P, callback: Q);
+
+    fn invoke_ask_password<'a, P: Into<Option<&'a Cancellable>>>(&self, password: &TlsPassword, cancellable: P) -> Result<TlsInteractionResult, Error>;
+
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn invoke_request_certificate<'a, P: IsA<TlsConnection>, Q: Into<Option<&'a Cancellable>>>(&self, connection: &P, flags: TlsCertificateRequestFlags, cancellable: Q) -> Result<TlsInteractionResult, Error>;
+
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn request_certificate<'a, P: IsA<TlsConnection>, Q: Into<Option<&'a Cancellable>>>(&self, connection: &P, flags: TlsCertificateRequestFlags, cancellable: Q) -> Result<TlsInteractionResult, Error>;
+
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn request_certificate_async<'a, P: IsA<TlsConnection>, Q: Into<Option<&'a Cancellable>>, R: FnOnce(Result<(), Error>) + Send + 'static>(&self, connection: &P, flags: TlsCertificateRequestFlags, cancellable: Q, callback: R);
+}
+
+impl<O: IsA<TlsInteraction>> TlsInteractionExt for O {
+    fn ask_password<'a, P: Into<Option<&'a Cancellable>>>(&self, password: &TlsPassword, cancellable: P) -> Result<TlsInteractionResult, Error> {
+        let cancellable = cancellable.into();
+        let cancellable = cancellable.to_glib_none();
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::g_tls_interaction_ask_password(self.to_glib_none().0, password.to_glib_none().0, cancellable.0, &mut error);
+            if error.is_null() { Ok(from_glib(ret)) } else { Err(from_glib_full(error)) }
+        }
+    }
+
+    fn ask_password_async<'a, P: Into<Option<&'a Cancellable>>, Q: FnOnce(Result<(), Error>) + Send + 'static>(&self, password: &TlsPassword, cancellable: P, callback: Q) {
```

Comment:
This must have `FnOnce(Result<TlsInteractionResult, Error>) ...` see `ask_password` return type,
so or ignore or make manual.
There will be also same error in others async functions for sure.

Reply:
Done